File: Demoblaze_Testing_TestNG_On_BrowserStack/src/main/java/com/example/pages/CartPage.java
package com.example.pages;

import com.example.config.Constants;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.FindBy;
import org.openqa.selenium.support.PageFactory;
import org.openqa.selenium.By;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;

import java.time.Duration;

public class CartPage {
    WebDriver driver;
    WebDriverWait wait;

    @FindBy(xpath = Constants.CART_LINK)
    WebElement cartLink;

    @FindBy(xpath = Constants.PLACE_ORDER_BUTTON)
    WebElement placeOrderBtn;

    public CartPage(WebDriver driver) {
        this.driver = driver;
        this.wait = new WebDriverWait(driver, Duration.ofSeconds(20));
        PageFactory.initElements(driver, this);
    }

    public void goToCart() {
        wait.until(ExpectedConditions.elementToBeClickable(cartLink));
        cartLink.click();
        wait.until(ExpectedConditions.elementToBeClickable(placeOrderBtn));
    }

    public void placeOrder() {
        wait.until(ExpectedConditions.elementToBeClickable(placeOrderBtn));
        placeOrderBtn.click();
        wait.until(ExpectedConditions.visibilityOfElementLocated(By.id(Constants.ORDER_MODAL_HEADER_ID)));
    }
}

File: Demoblaze_Testing_Pytest_On_BrowserStack/pages/cart_page.py
from selenium.webdriver.common.by import By
from selenium.webdriver.support.ui import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC
from config import CART_LINK, PLACE_ORDER_BUTTON

class CartPage:
    def __init__(self, driver):
        self.driver = driver
        self.wait = WebDriverWait(driver, 20)

    def go_to_cart(self):
        cart = self.wait.until(EC.element_to_be_clickable((By.XPATH, CART_LINK)))
        cart.click()

    def place_order(self):
        btn = self.wait.until(EC.element_to_be_clickable((By.XPATH, PLACE_ORDER_BUTTON)))
        btn.click()
